### PR TITLE
feat: Endpoint to complete an order in new Orders API

### LINKED ISSUE #6719: Endpoint for charging and fulfilling an order
Parent issue: #6302
 
API for charging/fulfilling an order which works with Payment SDKs on the server, calculated on the server taking in correct values for ticket prices, discount codes and tax

### PR DESCRIPTION
**Pull request by prateekj117**
Fixes #6719

### CODE REVIEW

**Comment by niranjan94** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 2
- Added 2
           

Complexity increasing per file
==============================
- app/api/custom/orders.py  10
- app/api/helpers/order.py  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/fossasia/open-event-orga-server/pullRequest?prid=4763653&bid=15881716)

File: app/api/custom/orders.py [hunk 1]
```diff
@@ -80,16 +84,129 @@ def resend_emails():
...
+        return make_response(jsonify(status='Order Unsuccessful', error='Tickets with id {} were not found in Event {}.'
+                                     .format(tickets_not_found, data['event_id'])), 404)
+    for ticket_info in ticket_list:
+        if (ticket_info.quantity - get_count(db.session.query(TicketHolder.id).filter_by(
+                ticket_id=int(ticket_info.id), deleted_at=None))) < quantity[ticket_info.id]:
+            return make_response(jsonify(status='Order Unsuccessful', error='Ticket already sold out.'), 409)
+    attendee_list = []
+    for ticket in tickets:
+        for ticket_amount in range(ticket['quantity']):
+            attendee = TicketHolder(**result[0], event_id=int(data['event_id']), ticket_id=int(ticket['id']))
```
Comment (niranjan94) [inline app/api/custom/orders.py]: Codacy Issue found: [invalid syntax](https://app.codacy.com/manual/fossasia/open-event-orga-server/pullRequest?prid=4763653)

File: app/api/custom/orders.py [hunk 2]
```diff
@@ -80,16 +84,129 @@ def resend_emails():
...
+    if tickets_not_found:
+        return make_response(jsonify(status='Order Unsuccessful', error='Tickets with id {} were not found in Event {}.'
+                                     .format(tickets_not_found, data['event_id'])), 404)
+    for ticket_info in ticket_list:
+        if (ticket_info.quantity - get_count(db.session.query(TicketHolder.id).filter_by(
+                ticket_id=int(ticket_info.id), deleted_at=None))) < quantity[ticket_info.id]:
+            return make_response(jsonify(status='Order Unsuccessful', error='Ticket already sold out.'), 409)
+    attendee_list = []
+    for ticket in tickets:
+        for ticket_amount in range(ticket['quantity']):
```
Comment (niranjan94) [inline app/api/custom/orders.py]: Codacy Issue found: [Unused variable 'ticket_amount'](https://app.codacy.com/manual/fossasia/open-event-orga-server/pullRequest?prid=4763653)

**Comment by niranjan94** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- app/api/custom/orders.py  8
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/fossasia/open-event-orga-server/pullRequest?prid=4763653&bid=15881716)

File: app/api/custom/orders.py [hunk 3]
```diff
@@ -122,14 +123,15 @@ def create_order():
                                      .format(tickets_not_found, data['event_id'])), 404)
     for ticket_info in ticket_list:
         if (ticket_info.quantity - get_count(db.session.query(TicketHolder.id).filter_by(
-                ticket_id=int(ticket_info.id), deleted_at=None))) < quantity[ticket_info.id]:
+            ticket_id=int(ticket_info.id), deleted_at=None))) < quantity[ticket_info.id]:
```
Comment (niranjan94) [inline app/api/custom/orders.py]: Codacy Issue found: [visually indented line with same indent as next logical line](https://app.codacy.com/manual/fossasia/open-event-orga-server/pullRequest?prid=4763653)

**Comment by niranjan94** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           

Complexity increasing per file
==============================
- app/api/custom/orders.py  12
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/fossasia/open-event-orga-server/pullRequest?prid=4763653&bid=15881716)

**Comment by niranjan94** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- app/api/custom/orders.py  14
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/fossasia/open-event-orga-server/pullRequest?prid=4763653&bid=15881716)

**Comment by niranjan94** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- app/api/custom/orders.py  14
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/fossasia/open-event-orga-server/pullRequest?prid=4763653&bid=15881716)

**Comment by niranjan94** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- app/api/custom/orders.py  14
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/fossasia/open-event-orga-server/pullRequest?prid=4763653&bid=15881716)

**Comment by iamareebjamal** [comment]
@prateekj117 Can this be completed soon?

**Comment by niranjan94** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- app/api/custom/orders.py  14
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/fossasia/open-event-orga-server/pullRequest?prid=4763653&bid=15881716)

**Comment by prateekj117** [comment]
@iamareebjamal Sure. Will look into it.

**Comment by niranjan94** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- app/api/custom/orders.py  15
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/fossasia/open-event-orga-server/pullRequest?prid=4763653&bid=15881716)

**Comment by niranjan94** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- app/api/custom/orders.py  20
         

Clones added
============
- app/api/custom/orders.py  5
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/fossasia/open-event-orga-server/pullRequest?prid=4763653&bid=15881716)

**Comment by niranjan94** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- app/api/custom/orders.py  20
         

Clones added
============
- app/api/custom/orders.py  5
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/fossasia/open-event-orga-server/pullRequest?prid=4763653&bid=15881716)

**Comment by niranjan94** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- app/api/custom/orders.py  20
         

Clones added
============
- app/api/custom/orders.py  5
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/fossasia/open-event-orga-server/pullRequest?prid=4763653&bid=15881716)

**Comment by niranjan94** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- app/api/custom/orders.py  20
         

Clones added
============
- app/api/custom/orders.py  5
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/fossasia/open-event-orga-server/pullRequest?prid=4763653&bid=15881716)

**Pull request merged**